router/core/context.go

```diff
@@ -253,11 +253,43 @@ type requestContext struct {
 	expressionContext expr.Context
 }
 
+func (c *requestContext) setErrorContext() {
+	if c.error == nil {
+		return
+	}
+	c.expressionContext.Request.Error = &ExprWrapError{c.error}
```

Comment:
Refer https://github.com/wundergraph/cosmo/pull/1545/files#r1936122019 and https://github.com/wundergraph/cosmo/pull/1545#discussion_r1936100140 for why we wrap